packages/field-base/src/slot-target-mixin.js

```diff
@@ -0,0 +1,114 @@
+import { dedupingMixin } from '@polymer/polymer/lib/utils/mixin.js';
+
+const SlotTargetMixinImplementation = (superclass) =>
+  class SlotTargetMixinClass extends superclass {
+    constructor() {
+      super();
+
+      /**
+       * @type {MutationObserver}
+       * @private
+       */
+      this.__slotTargetObserver = new MutationObserver(() => {
+        this._onSlotTargetContentChange();
+      });
+    }
+
+    /** @protected */
+    ready() {
+      super.ready();
+
+      if (this._slotTarget) {
+        // Observe for changes in the target element.
+        this.__slotTargetObserver.observe(this._slotTarget, {
+          childList: true,
+          characterData: true
```

Comment:
Add tests for the case when a text node is changed.

Reply:
Decided not to add this option at all, since the old `<vaadin-checkbox` implementation doesn't take care of such a case.